fast_llm/data/dataset/config.py

```diff
@@ -40,10 +40,14 @@ class SamplingData:
     cache_directory: pathlib.Path | None
     # TODO: This prevents the sampling config from being pickled in multiprocessing.
     distributed: "Distributed"
-    phase: PhaseType
+    dataset_name: str
     # Using a mutable rather than an int so it's shared with all copies made with `update`.
     _rank_counter: typing.Iterator[int] = itertools.count
 
+    def __post_init__(self):
```

Comment:
`__post_init__` shouldn't be implemented. You need to use `_validate` instead, though for this kind of backward compatibility fix `_from_dict` is more appropriate.